gensim/matutils.py

```diff
@@ -70,7 +70,7 @@ def argsort(x, topn=None):
             return numpy.argsort(x)[:topn]
         # numpy >= 1.8 has a fast partial argsort, use that!
         biggest = numpy.argpartition(x, topn)[:topn]
-        return biggest.take(numpy.argsort(x.take(biggest)))  # resort topn into order
```

Comment:
The -1 is needed to give the expected results when using numpy's argpartition and the below (minimal) changes to most_similar. (I did not test with bottleneck so didn't notice any API incompatability there.)